dotnet/samples/Concepts/Filtering/PIIDetectionWithFilters.cs

```diff
@@ -0,0 +1,404 @@
+        var result2 = await kernel.InvokePromptAsync("Hi, can you help me?");
+        logger.LogInformation("Result: {Result}", result2.ToString());
+
+        /*
+        Prompt: Hi, can you help me?
+        Result: Of course! I'm here to help. What do you need assistance with?
+        */
+    }
+
+    /// <summary>
+    /// Use Presidio Text Anonymizer to detect PII information in prompt and update the prompt by following specified rules before sending it to LLM.
+    /// Text Anonymizer API: https://microsoft.github.io/presidio/api-docs/api-docs.html#tag/Anonymizer.
+    /// </summary>
+    [Fact]
+    public async Task PromptAnonymizerAsync()
+    {
+        var builder = Kernel.CreateBuilder();
+
+        // Add Azure OpenAI chat completion service
+        builder.AddAzureOpenAIChatCompletion(
+            TestConfiguration.AzureOpenAI.ChatDeploymentName,
+            TestConfiguration.AzureOpenAI.Endpoint,
+            TestConfiguration.AzureOpenAI.ApiKey);
+
+        // Add logging
+        var logger = this.LoggerFactory.CreateLogger<PIIDetectionWithFilters>();
+        builder.Services.AddSingleton<ILogger>(logger);
+
+        // Add Microsoft Presidio Text Analyzer service and configure HTTP client for it. Text Analyzer results are required for Text Anonymizer input.
+        builder.Services.AddHttpClient<PresidioTextAnalyzerService>(client => { client.BaseAddress = new Uri("http://localhost:5001"); });
+
+        // Add Microsoft Presidio Text Anonymizer service and configure HTTP client for it
+        builder.Services.AddHttpClient<PresidioTextAnonymizerService>(client => { client.BaseAddress = new Uri("http://localhost:5002"); });
+
+        // Define anonymizer rules: redact phone number and replace person name with word "ANONYMIZED"
+        var anonymizers = new Dictionary<string, PresidioTextAnonymizer>
+        {
+            [AnalyzerEntityType.PhoneNumber] = new PresidioTextAnonymizer { Type = AnonymizerType.Redact },
+            [AnalyzerEntityType.Person] = new PresidioTextAnonymizer { Type = AnonymizerType.Replace, NewValue = "ANONYMIZED" }
+        };
+
+        // Add prompt filter to anonymize rendered prompt before sending it to LLM
+        builder.Services.AddSingleton<IPromptRenderFilter>(sp => new PromptAnonymizerFilter(
+            sp.GetRequiredService<ILogger>(),
+            sp.GetRequiredService<PresidioTextAnalyzerService>(),
+            sp.GetRequiredService<PresidioTextAnonymizerService>(),
+            anonymizers));
+
+        var kernel = builder.Build();
+
+        // Define instructions for LLM how to react when certain conditions are met for demonstration purposes
+        var executionSettings = new OpenAIPromptExecutionSettings
+        {
+            ChatSystemPrompt = "If prompt does not contain first and last name - return 'true'."
+        };
+
+        var result = await kernel.InvokePromptAsync("Hello world, my name is Jane Doe. My number is: 034453334", new(executionSettings));
```

Comment:
A more compelling demonstration of anonymizing peoples names would be for a case where you are uploading customer data. How about having a prompt which inserts data from a function which simulates reading customer data from a database and you want to anonymise the names and redact all of the phone numbers.